activities.el

```diff
@@ -809,11 +929,65 @@ activity's name is NAME."
-    (or (map-elt activities-activities name)
-        (make-activities-activity :name name))))
+  (pcase-let*
+      ((names (activities-names activities))
+       (max-age (activities--oldest-age activities))
+       (`(,old-col ,new-col ,blend-frac) activities-annotation-colors)
+       (prompt (format-prompt prompt default)))
+    (cl-labels
+	((activity-annotation-function (name)
+	   "Add buffer and file count, age, active and changed status to activity NAME."
+	   (when-let ((activity (map-elt activities-activities name)))
+	     (let (activity-data)
+	       (dolist (type '(last default))
+		 (when-let ((state (cl-struct-slot-value 'activities-activity type activity)))
+		   (let* ((time (map-elt (activities-activity-state-etc state) 'time))
+			  (buffers-and-files (activities--buffers-and-files state)))
+		     (setf (alist-get type activity-data)
+			   (list (and time (float-time (time-since time))) buffers-and-files)))))
+	       (pcase-let*
+		   ((`(,default-age ,default-buffers-and-files) (map-elt activity-data 'default))
+		    (`(,last-age ,last-buffers-and-files) (map-elt activity-data 'last)) ;possibly nil
+		    (age (if last-age (min last-age default-age) default-age))
+		    (buffers-and-files (if last-age last-buffers-and-files default-buffers-and-files))
+		    (num-buffers (length buffers-and-files))
+		    (num-files (seq-count #'stringp (mapcar #'cdr buffers-and-files)))
+		    (dirtyp (when last-buffers-and-files
+			      (activities--buffers-and-files-differ-p
+			       last-buffers-and-files
+			       default-buffers-and-files)))
+		    (annotation (format "%s%s buf%s %s file%s "
+					(if (activities-activity-active-p activity)
+					    (propertize "@" 'face 'bold) " ")
+					(propertize (format "%2d" num-buffers) 'face 'success)
+					(if (= num-buffers 1) " " "s")
+					(propertize (format "%2d" num-files) 'face 'warning)
+					(if (= num-files 1) " " "s")))
+		    (age-color  (apply #'color-rgb-to-hex
+				       (cl-loop for co in (color-name-to-rgb old-col)
+						for cn in (color-name-to-rgb new-col)
+						for cd in (color-name-to-rgb (face-foreground 'default))
+						collect (+ (* blend-frac (+ cn (* (- co cn) (/ age max-age))))
+							   (* (- 1. blend-frac) cd)))))
+		    (age-annotation (propertize
+				     (format "%10s" (activities--age age))
+				     'face `(:foreground ,age-color :weight bold)))
+		    (dirty-annotation (if dirtyp (propertize "*" 'face 'bold) " ")))
+		 (concat (propertize " " 'display
+				     `(space :align-to (- right ,(+ 1 (length annotation)
+								    (length age-annotation)))))
+			 annotation age-annotation dirty-annotation)))))
+	 (activity-table (str pred action)
+	   "Complete activities from STR, using completion PRED and ACTION."
+	   (if (eq action 'metadata)
+	       `(metadata (annotation-function . ,#'activity-annotation-function)
+			  ,@(when activities-sort-by
+			      `(,(cons 'display-sort-function activities-sort-by))))
```

Comment:
Is it necessary to support sorting being optional?  It seems like disabling sorting would be of no benefit.

Reply:
I guess no-sorting is actually "sorting by the order I made these activities", right?  Could be useful to someone.

Reply:
I don't think the order of the `activities-activities` list is defined, so "no sorting" would be equivalent to "unknown order," which doesn't seem useful to me.  Unless there's a potential performance issue that would make it necessary to disable sorting, I think I'd rather go for the simpler code that expects the sort-by function to always be set (it could be set to `always` or `ignore` to effectively disable sorting, if the user really wanted).

Reply:
Fair enough, the user can always slot `#'ignore` in if they want no change.  I also found a last minute slight flaw in the sort logic and fixed that.